packages/runtime/src/binding/binding-behavior.ts

```diff
@@ -1,16 +1,24 @@
-import { Constructable, IContainer, Registration, Writable } from '@aurelia/kernel';
+import { Constructable, Decoratable, Decorated, IContainer, Registration, Writable } from '@aurelia/kernel';
+import { IScope, LifecycleFlags } from '../observation';
 import { IResourceKind, IResourceType } from '../resource';
+import { IBinding } from './binding';
+
+export interface IBindingBehavior {
+  bind(flags: LifecycleFlags, scope: IScope, binding: IBinding): void;
+  unbind(flags: LifecycleFlags, scope: IScope, binding: IBinding): void;
+}
 
 export interface IBindingBehaviorSource {
   name: string;
 }
 
-export type IBindingBehaviorType = IResourceType<IBindingBehaviorSource>;
+export interface IBindingBehaviorType extends IResourceType<IBindingBehaviorSource> {
+}
+
+export type BindingBehaviorDecorator = <T extends Constructable>(target: Decoratable<IBindingBehavior, T>) => Decorated<IBindingBehavior, T> & IBindingBehaviorType;
 
-export function bindingBehavior(nameOrSource: string | IBindingBehaviorSource): <T extends Constructable>(target: T) => T & IResourceType<IBindingBehaviorSource> {
-  return function<T extends Constructable>(target: T): T & IResourceType<IBindingBehaviorSource> {
-    return BindingBehaviorResource.define(nameOrSource, target);
-  };
+export function bindingBehavior(nameOrSource: string | IBindingBehaviorSource): BindingBehaviorDecorator {
```

Comment:
```suggestion
export function bindingBehavior(name: string): BindingBehaviorDecorator;
export function bindingBehavior(source: IBindingBehaviorSource): BindingBehaviorDecorator;
export function bindingBehavior(nameOrSource: string | IBindingBehaviorSource): BindingBehaviorDecorator {
```
Since you mentioned you'd be using it as a template for other decorators, I think we should include overload signatures right away (unless that introduces new linting errors). I believe this will make the APIs clearer and easier to document properly.

Reply:
It does introduce a new linting error on `unified-signatures` 😞. 

I like the verbose API documentation though, so it might be worth suppressing?

Reply:
Let's just disable that silly rule.

Reply:
Just here? Or global? I'll assume the latter. 😊